dataland-backend/src/main/kotlin/org/dataland/datalandbackend/controller/DataController.kt

```diff
@@ -76,17 +91,18 @@ abstract class DataController<T>(
     }
 
     override fun getCompanyAssociatedData(dataId: String): ResponseEntity<CompanyAssociatedData<T>> {
-        val companyId = dataMetaInformationManager.getDataMetaInformationByDataId(dataId).company.companyId
+        val metaInfo = dataMetaInformationManager.getDataMetaInformationByDataId(dataId)
+        if (!isDataViewableByUser(metaInfo, DatalandAuthentication.fromContextOrNull())) {
+            throw InvalidInputApiException("Access denied", "You are trying to access a unreviewed dataset")
```

Comment:
shouldn't we return 401 forbidden in this case?

Reply:
I tried to find an argument for this exception, because I saw a reason for doing so. But I can not find a good one. Changed.

Reply:
ok